src/plugins/recursion/README.md

```diff
@@ -81,10 +81,186 @@ kdb set /recursive/define/#2/ref/#1 /recursive/define/#5
 Arbitrary additional data can be saved too such as:
 `kdb set /recursive/define/#2/permission admin`
 
+## A more real example
+
+In order to better show how this plugin can be used, a more sophisticated example will be given.
+Imagine that you want to create your own menu with arbitrary submenus possible. The following example
+is inspired from libreoffice and simplified.
+
+Imagine an editor with a `File` menu with a `Print` option as well as a submenu called `Settings`.
+The `Settings` submenu has `Global Settings` and  `Autocorrect Options`
+Beside the `File` menu we also have a `Tool` menu which also has the same `Settings` submenu but also
+has gotten a `Extension Manager` and a Menu `Macros`. Macros has again a submenu `Organize Macros` which
+again is a menu and has `Macro Basics`.
+
+The following picture illustrates the menu tree:
+![menu-example](example.png?raw=true "Example")
+
+So how can we map this into an elektra configuration?
+
+First we create a config and mount it. Next we will set the specification for all menu entries as following:
+```
+kdb mount config.dump /editor dump recursion
+sudo kdb setmeta /editor check/recursion/vertex "menu"
+sudo kdb setmeta /editor check/recursion/ref "menuref"
+```
+Each key definition in the form of `/editor/menu` will now be treated as recursive entry. You can define arbitrary
+many menues like this: `/editor/menu/#[d]` where `[d]` stands for a positive number.
+
+Menues can be linked via `menuref` like this: `/editor/menu/#[d]/menuref/#[d]` and is again in the array notation.
+Note that `menu` and `menuref` are our defined keywords in the setmeta command from above.
```

Comment:
That should be said earlier. First explain specification or configuration, do not mix up.